hpedockerplugin/hpe/hpe_3par_common.py

```diff
@@ -938,3 +967,199 @@ def _get_prioritized_host_on_3par(self, host, hosts, hostname):
+                self._copy_volume(src_3par_vol_name, dst_3par_vol_name,
+                                  cpg=cpg, snap_cpg=snap_cpg,
+                                  tpvv=tpvv, tdvv=tdvv,
+                                  compression=compression)
+
+                # Check if flash cache needs to be enabled
+                flash_cache = \
+                    self.get_flash_cache_policy(src_vref['flash_cache'])
+
+                if flash_cache is not None:
+                    try:
+                        self._add_volume_to_volume_set(dst_volume,
+                                                       src_3par_vol_name,
+                                                       cpg, flash_cache)
+                    except exception.InvalidInput as ex:
+                        # Delete volume if unable to add it to volume set
+                        self.client.deleteVolume(dst_3par_vol_name)
+                        LOG.error(_LE("Exception: %s"), ex)
+                        raise exception.PluginException(ex)
+            else:
+                # The size of the new volume is different, so we have to
+                # copy the volume and wait.  Do the resize after the copy
+                # is complete.
+                LOG.debug("Creating a clone of volume, using offline copy.")
+
+                # we first have to create the destination volume
+                model_update = self.create_volume(dst_volume)
+
+                optional = {'priority': 1}
+                body = self.client.copyVolume(src_3par_vol_name,
+                                              dst_3par_vol_name, None,
+                                              optional=optional)
+                task_id = body['taskid']
+
+                task_status = self._wait_for_task_completion(task_id)
+                if task_status['status'] is not self.client.TASK_DONE:
+                    dbg = {'status': task_status, 'id': dst_volume['id']}
+                    msg = _('Copy volume task failed: create_cloned_volume '
+                            'id=%(id)s, status=%(status)s.') % dbg
+                    raise exception.PluginException(msg)
+                else:
+                    LOG.debug('Copy volume completed: create_cloned_volume: '
+                              'id=%s.', dst_volume['id'])
+
+                return model_update
+
+        except hpeexceptions.HTTPForbidden:
+            raise exception.NotAuthorized()
+        except hpeexceptions.HTTPNotFound:
+            raise exception.NotFound()
+        except Exception as ex:
+            LOG.error("Exception: %s", ex)
+            raise exception.PluginException(ex)
+
+    def _copy_volume(self, src_name, dest_name, cpg, snap_cpg=None,
+                     tpvv=True, tdvv=False, compression=None):
+        # Virtual volume sets are not supported with the -online option
+        LOG.debug('Creating clone of a volume %(src)s to %(dest)s.',
```

Comment:
you can make to LOG.info() since for code tracing it will help if the logging is restrictive.